### ISSUE TITLE
Disable captcha storefont config still load captcha js files

### ISSUE THREAD

**Issue by ia-gaurav** [magento/magento2#32987]
**Preconditions (*)**

- Magento Version - 2.4.2
- Default Magento Theme
- Captcha is disabled on Frontend 

**Steps to reproduce (*)**
- Do fresh Magento install - with or without demo Data
- Open developer tool, goto network tab
- Here you will see, unwanted js call for captcha  
![image](https://user-images.githubusercontent.com/63009790/118082256-79a48680-b3da-11eb-9966-b6cf8286cd21.png)

**Expected result (*)**
- No Captcha related scripts should be called 

**My expected** 

- we don't need disable completely captcha module but if admin owner set config disable in backend both captcha blocks and js related may not visible in frontend

**Actual result (*)**
- 11 extra request of size 17.2 kB is getting called on page

**Comment by mrtuvn** [comment]
Hi i don't think you can disable captcha module since this module dependency magento checkout. Which mean disable captcha may break checkout function

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Captcha/etc/module.xml

**Comment by ia-gaurav** [comment]
yes but issue is that enabling captcha cause extra 5 unwanted request call in network even when disabled from admin 

![image](https://user-images.githubusercontent.com/63009790/117695803-42698600-b1de-11eb-8281-84ee7047f89d.png)

**Comment by mrtuvn** [comment]
maybe we still have issue and now time for investigate more on that

**Comment by hostep** [comment]
@mrtuvn:

> Hi i'm not sure how do you disable captcha module since this module dependency magento checkout. If completely disable module may break checkout function.

This is incorrect, the dependency works the other way, it means: disabling the checkout may break the captcha module (even though this sounds unlikely). Also, the dependencies defined in the `module.xml` file are soft dependencies, so we can't really rely on that information.

**Comment by ia-gaurav** [comment]
i donot want unwated netwrok call of Captcha when it is disabled, any way for that

**Comment by mrtuvn** [comment]
Hi @ia-gaurav Can you provide more steps for check the case unwanted load captcha after disable config from backend

**Comment by ia-gaurav** [comment]
@magento give me 2.4-develop instance

**Comment by ia-gaurav** [comment]
@mrtuvn

See below screenshot , no captha enabled on frontend site too many unwated calls - 11 extra request of size 17.2 kB

![image](https://user-images.githubusercontent.com/63009790/117958111-54f7d280-b338-11eb-8f85-052958c23986.png)

**Comment by mrtuvn** [comment]
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Captcha/view/frontend/layout/default.xml#L17
Here is the magic happen!
No matter you enable or disable this component always loaded

**Comment by mrtuvn** [comment]
Maybe js problem should report in another seperate ticket
May I ask you update description? Pre-condition, Steps check, expect, actually result

**Comment by ia-gaurav** [comment]
> Maybe js problem should report in another seperate ticket
> May I ask you update description? Pre-condition, Steps check, expect, actually result

ok let me update it

**Comment by mrtuvn** [comment]
Should we update title for this issue ?
Example "Disable captcha storefont config still load captcha js files"
My expected we don't need disable completely captcha module but if admin owner set config in backend both captcha blocks and js related may not visible in frontend

**Comment by ia-gaurav** [comment]
yes please do

**Comment by magento-engcom-team** [comment]
@mrtuvn Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @mrtuvn 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-42316` were created

**Issue Available**: @mrtuvn, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by engcom-Delta** [comment]
Hi @ia-gaurav ,

Thanks for your reporting and collaboration.
We have verified the issue in latest 2.4-develop instance and the issue is reproducible.
Kindly refer the screenshots.

**Steps to Reproduce**

1. Install 2.4-develop instance
2. Disable captcha on frontend admin
![image](https://github.com/user-attachments/assets/2801cc30-f778-499b-8dce-fd5f21357b38)
3. Access frontend and check network tab.
4. Observe js call for captcha is displaying
<img width="960" alt="captcha js" src="https://github.com/user-attachments/assets/27e6f5e8-6936-48e7-9196-d3d3efc9c43b">

Thanks.
